apps/address-book/app/components/Panel/NewEntity.js

```diff
@@ -28,29 +30,44 @@ class NewEntity extends React.Component {
   changeType = type => {
     this.setState({
       type: ENTITY_TYPES[type],
+      isCustomType: ENTITY_TYPES.length - 1,
     })
   }
 
   handleSubmit = () => {
-    const { name, address, type } = this.state
+    const { name, address, type, isCustomType, customTypeName } = this.state
     const error = {}
     if (!name) {
       error.name = 'Please provide a name'
     }
     if (!web3Utils.isAddress(address)) {
       error.address = 'Please provide a valid ethereum address'
     }
+    if (isCustomType && !customTypeName) {
+      error.customTypeName = 'Please provide a custom type name'
+    }
 
     if (Object.keys(error).length) {
       this.setState({ error: error })
     } else {
       this.setState(INITIAL_STATE)
-      this.props.onCreateEntity({ name, address, type })
+      this.props.onCreateEntity({
+        name,
+        address,
+        type: isCustomType ? customTypeName : type
```

Comment:
Rather than saving this as a simple `type` string on the entity, we probably want a separate api call for `createType`, where we pass it the name as well as the `bg` and `fg` colors. Then, when the result of that save comes back, we can show the new type in the drop-down.

However, implementing this requires not just code changes, but a UX/design discussion. I think we should close this PR for now, since the final version will probably look quite different, and circle back to this improvement once all of our apps are on mainnet.